app/webpacker/components/Tickets/TicketHeader/StatusView.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { Button, Dropdown, Icon } from 'semantic-ui-react';
+import _ from 'lodash';
+import { actionUrls } from '../../../lib/requests/routes.js.erb';
+import { ticketStatuses } from '../../../lib/wca-data.js.erb';
+import useSaveAction from '../../../lib/hooks/useSaveAction';
+import Loading from '../../Requests/Loading';
+import useInputState from '../../../lib/hooks/useInputState';
+
+const ticketTypePrefix = 'Tickets';
+
+function StatusViewEditMode({
+  ticketDetails, currentStakeholder, setEditMode, sync,
+}) {
+  // metadataType will be a string like TicketsTicketType
+  const { ticket: { id, metadata, metadata_type: metadataType } } = ticketDetails;
+  // To get ticketType, first remove prefix 'Tickets' and then convert it to snake case.
+  const ticketType = _.snakeCase(_.replace(metadataType, ticketTypePrefix, ''));
+
+  const [newStatus, setNewStatus] = useInputState(metadata.status);
+  const { save, saving } = useSaveAction();
+
+  function saveStatus(status) {
+    save(
+      actionUrls.tickets.updateStatus(id),
+      {
+        ticket_status: status,
+        acting_stakeholder_id: currentStakeholder.id,
+      },
+      sync,
+      { method: 'POST' },
+    );
+  }
+
+  if (saving) {
+    return <Loading />;
+  }
+
+  return (
+    <>
+      <span>{'Status: '}</span>
+      <Dropdown
+        inline
+        options={Object.keys(ticketStatuses[ticketType]).map((key) => ({
+          key,
+          text: ticketStatuses[ticketType][key],
+          value: key,
+        }))}
```

Comment:
Not a fan of computing these inline. Move to a separate `useMemo` please :upside_down_face:

Reply:
Done

Reply:
Since this is minor front-end only change, I think it's fine to go ahead without Gregor's approval. The merging is blocked because Gregor has 'requested for a change'. I think only option for me is to skip Gregor's approval is to dismiss the review. Since I have already actioned what Gregor requested, I'm going ahead with dismissing Gregor's review. Hope this works.